### ISSUE TITLE
iPad Version?

### ISSUE THREAD

**Issue by aadrian** [mfikes/replete#80]
Any chance making an iPad version of the app, optimized for bigger screens?

Thank you.

**Comment by mfikes** [comment]
Replete actually does have (minimal) support for iPad. For example, it isn't just the iPhone UI at 2x. It supports the iPad keyboard, and things like `doc` don't reflow like on iPhone, etc.

**Comment by aadrian** [comment]
> Replete actually does have (minimal) support for iPad. 

Since the iPads all have enough screen estate (usually as big as the most terminal windows), maybe Plank would be a better alternative for a "RepleteHD" version ? ( I would buy one :). )

> It supports the iPad keyboard

I haven't observed this, as the keys look for me just like huge scaled iPhone keys.
The main problem is however the missing special keys from the layout (the constant switch to keyboard modes make it useless).
E.g. how RealVNC for iOS des this:
http://wpuploads.appadvice.com/wp-content/uploads/2012/05/vncviewer_v203_ipad2_screen1large-642x481.jpg

On iPadPro is even worse - there would be enough screen estate for a full IDE :).

Of course, for the iPhone, is not really useful than to try a few things, but a more functional iPad version could be a fantastic learning companion, since tablets are used allot as a learning devices.

**Comment by mfikes** [comment]
@aadrian Yes, the iPad screen real estate could be better used. Ideas and patches welcome. 

(Since there are multiple open-source contributors, I have no plans to ever make a paid version.)

With respect to scaled keys, it is definitely an iPad UI. It can go landscape, for example.

http://a4.mzstatic.com/us/r30/Purple7/v4/8f/b6/db/8fb6db71-e34b-c3fd-1a59-dfcc95da35c4/screen480x480.jpeg

Perhaps you are using it on an iPad Pro and it is scaling up the older UI that was released before this newer device came out. There is a new release coming out soon which I added an iPad Pro icon—perhaps that new binary will scale properly on this device.

With respect to special keys, there is an issue for this: #62 (The new release makes use of Parinfer which will alleviate this issue somewhat. See http://blog.fikesfarm.com/posts/2015-11-20-parinfer-in-replete.html)

So, for iPad-specific features, we'd need specific ideas to act upon.

**Comment by aadrian** [comment]
@mfikes Thank you for the details.

Since tablets could be used more for learning (and there's enough screen estate), IMHO the following would greatly help this process:
 - interactive feature tour for new users. Something like: https://usablica.github.io/intro.js/ 
 - cheatsheet integration, or at least linking (integration would allow more control)
 - special key lines, despite Parinfer (to avoid switching)
 - Input text box, instead of text line
 - settings (persistent) - at least font and size
 - this would also greatly help new users https://github.com/mfikes/planck/issues/241
 - selection for copy and paste is horrible on iOS devices. Maybe Parinfer could help to correctly copy stuff from examples to quickly try them in the REPL.
